### PR TITLE
d_a_obj_tumble_weed mostly done

### PR DESCRIPTION
**Pull request by elijah-thomas774**
~~`calcMatrix` is a mess. Quatf stuff and I dont want to figure that out~~

`calcTimer<u16>` template weak func is out of order with state `execute` weak func: see #89 

`adjustSpeed` has ~~some small issues~~ regswap

### CODE REVIEW

File: include/s/s_Math.h [hunk 1]
```diff
@@ -22,6 +22,14 @@ void addCalcScaled(float *value, float stepSize, float maxStep);
 int absDiff(short a1, short a2);
 BOOL chaseUC(u8 *value, u8 target, u8 stepSize);
 
+template <typename T>
```
Comment (robojumper) [inline include/s/s_Math.h:25]: This function already exists in c_math. If this is a better place for it we should remove the other one.

Comment (elijah-thomas774) [reply include/s/s_Math.h:25]: Understood. nsmbw has it named in sLib

**Comment by elijah-thomas774** [comment]
this also removes d_t_tumble_weed's decr function. I opted to remove it due to it not being linked in anyway

File: src/REL/d/a/obj/d_a_obj_tumble_weed.cpp [hunk 2]
```diff
@@ -1,13 +1,397 @@
...
+    cM3dGPla pla;
+
+    const f32 a = mVec3_c::Ey.dot(pla.mNormal);
+    const f32 b = mVec3_c::Ey.cross(pla.mNormal).length();
+
+    mAng ang = mAng::fromRad(EGG::Math<f32>::abs(EGG::Math<f32>::atan2(b, a)));
+
+    dBgS::GetInstance()->GetTriPla(mObjAcch.mGnd, &pla);
+
+    f32 tmp2 = nw4r::math::CosIdx(ang) * 0.5f;
```
Comment (robojumper) [inline src/REL/d/a/obj/d_a_obj_tumble_weed.cpp:327]:
```suggestion
    f32 t = 0.5f;
    f32 tmp2 = ang.cos() * t;
```
works a bit better

**Pull request merged**